pkg/controller/servingcert/controller/secret_updating_controller.go

```diff
@@ -58,9 +58,16 @@ func (sc *serviceServingCertUpdateController) addSecret(obj metav1.Object) bool
 }
 
 func (sc *serviceServingCertUpdateController) updateSecret(old, cur metav1.Object) bool {
-	// if the current doesn't have a service name, check the old
+	curSecret := cur.(*v1.Secret)
+	oldSecret := old.(*v1.Secret)
+	// if the current data does not match or doesn't have a service name, check the old
 	// TODO drop this
-	return sc.addSecret(cur) || sc.addSecret(old)
+	if string(curSecret.Data[v1.TLSCertKey]) == string(oldSecret.Data[v1.TLSCertKey]) &&
```

Comment:
This seems right to me but my brain wants to think about the inverse case (i.e., cur and old data do not match or cur does not have 2 items, so add old). Not sure if that is clearer to anyone else :)

Reply:
+1, can't explain why my brain wants it that way.

Reply:
lol ok, 2 against 1 :)